# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/tabs/src/sync/bridge.rs
# Commit: 57b2cda7e0db
# Full Hash: 57b2cda7e0db8b07fcc1e09113ce26add178c68a
# Author: Sammy Khamis <skhamis@mozilla.com>
# Date: 2023-03-01 03:56:13
# Regressor Bug: 1818349
# File Overlap Count: 23
# Overlapping Files: third_party/rust/sync15/src/client/coll_update.rs, third_party/rust/sync15/src/client/mod.rs, third_party/rust/sync15/.cargo-checksum.json, third_party/rust/sync15/src/client/sync.rs, Cargo.toml
# Description:
#   Bug 1818349 Part 1: Vendor new version of application-services r=markh
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D171181
# ==============================================================================

diff -r eaf9f12a1481 -r 57b2cda7e0db third_party/rust/tabs/src/sync/bridge.rs
--- a/third_party/rust/tabs/src/sync/bridge.rs	Tue Feb 28 04:22:35 2023 +0000
+++ b/third_party/rust/tabs/src/sync/bridge.rs	Tue Feb 28 05:14:29 2023 +0000
@@ -4,7 +4,7 @@
 
 use std::sync::{Arc, Mutex};
 
-use crate::error::{ApiResult, Result, TabsApiError};
+use crate::error::{ApiResult, TabsApiError};
 use crate::sync::engine::TabsSyncImpl;
 use crate::TabsStore;
 use error_support::handle_error;
@@ -47,7 +47,7 @@
 impl BridgedEngine for BridgedEngineImpl {
     type Error = TabsApiError;
 
-    #[handle_error]
+    #[handle_error(crate::Error)]
     fn last_sync(&self) -> ApiResult<i64> {
         Ok(self
             .sync_impl
@@ -58,7 +58,7 @@
             .as_millis())
     }
 
-    #[handle_error]
+    #[handle_error(crate::Error)]
     fn set_last_sync(&self, last_sync_millis: i64) -> ApiResult<()> {
         self.sync_impl
             .lock()
@@ -67,7 +67,7 @@
         Ok(())
     }
 
-    #[handle_error]
+    #[handle_error(crate::Error)]
     fn sync_id(&self) -> ApiResult<Option<String>> {
         Ok(match self.sync_impl.lock().unwrap().get_sync_assoc()? {
             EngineSyncAssociation::Connected(id) => Some(id.coll.to_string()),
@@ -75,7 +75,7 @@
         })
     }
 
-    #[handle_error]
+    #[handle_error(crate::Error)]
     fn reset_sync_id(&self) -> ApiResult<String> {
         let new_id = SyncGuid::random().to_string();
         let new_coll_ids = CollSyncIds {
@@ -89,7 +89,7 @@
         Ok(new_id)
     }
 
-    #[handle_error]
+    #[handle_error(crate::Error)]
     fn ensure_current_sync_id(&self, sync_id: &str) -> ApiResult<String> {
         let mut sync_impl = self.sync_impl.lock().unwrap();
         let assoc = sync_impl.get_sync_assoc()?;
@@ -105,7 +105,7 @@
         Ok(sync_id.to_string()) // this is a bit odd, why the result?
     }
 
-    #[handle_error]
+    #[handle_error(crate::Error)]
     fn prepare_for_sync(&self, client_data: &str) -> ApiResult<()> {
         let data: ClientData = serde_json::from_str(client_data)?;
         self.sync_impl.lock().unwrap().prepare_for_sync(data)
@@ -116,14 +116,14 @@
         Ok(())
     }
 
-    #[handle_error]
+    #[handle_error(crate::Error)]
     fn store_incoming(&self, incoming: Vec<IncomingBso>) -> ApiResult<()> {
         // Store the incoming payload in memory so we can use it in apply
         *(self.incoming.lock().unwrap()) = incoming;
         Ok(())
     }
 
-    #[handle_error]
+    #[handle_error(crate::Error)]
     fn apply(&self) -> ApiResult<ApplyResults> {
         let mut incoming = self.incoming.lock().unwrap();
         // We've a reference to a Vec<> but it's owned by the mutex - swap the mutex owned
@@ -141,7 +141,7 @@
         })
     }
 
-    #[handle_error]
+    #[handle_error(crate::Error)]
     fn set_uploaded(&self, server_modified_millis: i64, ids: &[SyncGuid]) -> ApiResult<()> {
         self.sync_impl
             .lock()
@@ -149,13 +149,13 @@
             .sync_finished(ServerTimestamp::from_millis(server_modified_millis), ids)
     }
 
-    #[handle_error]
+    #[handle_error(crate::Error)]
     fn sync_finished(&self) -> ApiResult<()> {
         *(self.incoming.lock().unwrap()) = Vec::default();
         Ok(())
     }
 
-    #[handle_error]
+    #[handle_error(crate::Error)]
     fn reset(&self) -> ApiResult<()> {
         self.sync_impl
             .lock()
@@ -164,7 +164,7 @@
         Ok(())
     }
 
-    #[handle_error]
+    #[handle_error(crate::Error)]
     fn wipe(&self) -> ApiResult<()> {
         self.sync_impl.lock().unwrap().wipe()?;
         Ok(())
@@ -210,7 +210,7 @@
     }
 
     // Decode the JSON-encoded IncomingBso's that UniFFI passes to us
-    #[handle_error]
+    #[handle_error(crate::Error)]
     fn convert_incoming_bsos(&self, incoming: Vec<String>) -> ApiResult<Vec<IncomingBso>> {
         let mut bsos = Vec::with_capacity(incoming.len());
         for inc in incoming {
@@ -220,7 +220,7 @@
     }
 
     // Encode OutgoingBso's into JSON for UniFFI
-    #[handle_error]
+    #[handle_error(crate::Error)]
     fn convert_outgoing_bsos(&self, outgoing: Vec<OutgoingBso>) -> ApiResult<Vec<String>> {
         let mut bsos = Vec::with_capacity(outgoing.len());
         for e in outgoing {
@@ -279,7 +279,7 @@
                 title: "my first tab".to_string(),
                 url_history: vec!["http://1.com".to_string()],
                 icon: None,
-                last_used: 0,
+                last_used: 2,
             },
             RemoteTab {
                 title: "my second tab".to_string(),